internal/grpcutil/method_test.go

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package grpcutil
+
+import (
+	"testing"
+)
+
+func TestParseMethod(t *testing.T) {
+	testCases := []struct {
+		methodName  string
+		wantService string
+		wantMethod  string
+		wantError   bool
+	}{
+		{methodName: "/s/m", wantService: "s", wantMethod: "m", wantError: false},
+		{methodName: "/p.s/m", wantService: "p.s", wantMethod: "m", wantError: false},
+		{methodName: "/p/s/m", wantService: "p/s", wantMethod: "m", wantError: false},
+		{methodName: "/", wantError: true},
+		{methodName: "/sm", wantError: true},
+		{methodName: "", wantError: true},
+		{methodName: "sm", wantError: true},
+	}
+	for _, tc := range testCases {
+		s, m, err := ParseMethod(tc.methodName)
+		if (err != nil) != tc.wantError || s != tc.wantService || m != tc.wantMethod {
+			t.Errorf("ParseMethod(%s) = (%s, %s, %v), want (%s, %s, %v)", tc.methodName, s, m, err, tc.wantService, tc.wantMethod, tc.wantError)
+		}
+		if s != tc.wantService || m != tc.wantMethod {
+			t.Errorf("Parseing %q got service %q, method %q, want service %q, method %q",
+				tc.methodName, s, m, tc.wantService, tc.wantMethod,
+			)
+		}
+	}
```

Comment:
This block is not required now, since the above check covers everything, right? Or am I missing something?